### ISSUE TITLE
Script in `build_transfer_tx` should be empty 

### ISSUE THREAD

**Issue by digorithm** [FuelLabs/fuels-rs#630]
It was mentioned by @adlerjohn, in our internal conversations that an empty script should still be a valid script. The transfer script being built [here](https://github.com/FuelLabs/fuels-rs/blob/4f154c77f83ca62c9b7be4bb49b2422836ac9a99/packages/fuels-signers/src/wallet.rs#L281) isn't empty; it contains only a `ret(1)`. The rationale behind it being `ret(1)` _seems_ to be lost. Removing the line linked and effectively passing `script: vec![]` causes some tests to fail. 

We should investigate why that is the case and try and make it work with an empty script, if possible. For more motivation behind it and/or more explanation, reach out to @adlerjohn.

**Comment by iqdecay** [comment]
I think this could go hand in hand with the general consolidation of our functions and methods surrounding scripts (that @MujkicA mentioned in another PR).

**Comment by iqdecay** [comment]
After some exploration, the reason for this is pretty obscure: the transactions revert with  an `ErrorFlag` error. I'm digging further into this.

**Comment by iqdecay** [comment]
Unblocked by https://github.com/FuelLabs/fuel-core/pull/825

**Comment by hal3e** [comment]
My guess is that when you are passing an empty script the predicate data offset is wrong. Which will definitely make the predicate fail if you use dynamic data. In addition, if other predicate tests are passing and the vector one is failing then this is most likely the issue.
